File: clients/js/src/features/uri.ts
import { Input } from "../input";

import { Uri } from "@polywrap/client-js";

export function runTestCase(input: unknown): void {
  const str = Input.expectString(input);

  const uri = new Uri(str);

  console.log("WRAP URI successfully created.");
  console.log(`uri - ${uri}`);
  console.log(`uri.authority - ${uri.authority}`);
  console.log(`uri.path - ${uri.path}`);
}

File: clients/py/src/features/uri.py
from typing import Any

from validators import validate_uri


def run_test_case(input: Any) -> None:
    try:
        uri = validate_uri(input)

        print("WRAP URI successfully created.")
        print(f"uri - {uri}")
        print(f"uri.authority - {uri.authority}")
        print(f"uri.path - {uri.path}")
    except Exception as e:
        raise ValueError(f"Invalid URI Received: {input}") from e
